geopandas/tests/test_geodataframe.py

```diff
@@ -772,6 +772,27 @@ def test_dataframe_to_geodataframe(self):
         with pytest.raises(ValueError):
             df.set_geometry("location", inplace=True)
 
+    def test_dataframe_not_manipulated(self):
+        df = pd.DataFrame(
+            {
+                "A": range(len(self.df)),
+                "latitude": self.df.geometry.centroid.y,
+                "longitude": self.df.geometry.centroid.x,
+            },
+            index=self.df.index,
+        )
+        df_copy = df.copy()
+        gf = GeoDataFrame(
+            df,
+            geometry=points_from_xy(df["longitude"], df["latitude"]),
+            crs=self.df.crs,
+        )
+        assert isinstance(df, pd.DataFrame)
+        assert "geometry" not in df
+        pd.testing.assert_frame_equal(df, df_copy)
```

Comment:
nit, assert_frame_equal is already imported in test_geodataframe
```suggestion
    assert_frame_equal(df, df_copy)
```

Reply:
thanks! changed in https://github.com/geopandas/geopandas/pull/2667/commits/ccfb62a5f4904226c2394ee70f5264bfec9d71ee